File: JavaScript_Courses/JavaScript_Basics/Preparation/05_While_Loop_Exercise/02_Exam_Preparation.js
function examPreparation(input) {
    let limitBadGrades = Number(input[0]);
    let i = 1;

    let currentBadGrades = 0;
    let sumScore = 0;
    let problemsCount = 0;
    let lastProblem = "";

    while (true) {
        let currentTask = input[i];
        let currentGrade = Number(input[i + 1]);


        if (currentTask == "Enough") {
            console.log(`Average score: ${(sumScore / problemsCount).toFixed(2)}`);
            console.log(`Number of problems: ${problemsCount}`);
            console.log(`Last problem: ${lastProblem}`);
            break;
        }

        if (currentGrade <= 4) {
            currentBadGrades++;
            if (currentBadGrades == limitBadGrades) {
                console.log(`You need a break, ${currentBadGrades} poor grades.`);
                break;
            }
        }

        lastProblem = currentTask;
        problemsCount++;
        sumScore += currentGrade;
        i += 2;
    }

}


// examPreparation([
//     "3",
//     "Money",
//     "6",
//     "Story",
//     "4",
//     "Spring Time",
//     "5",
//     "Bus",
//     "6",
//     "Enough"
// ]);


// examPreparation([
//     "2",
//     "Income",
//     "3",
//     "Game Info",
//     "6",
//     "Best Player",
//     "4"
// ]);

File: Python_Courses/Python_Basics/Python_PB_2023_10/05_While_Loop_Exercise/02_Exam_Preparation.py
bad_grades_threshold = int(input())
total_score = 0
problems_count = 0
last_problem = ""
bad_grades_count = 0

# is_positive_result = False

command = input()
while command != "Enough":
    problem_name = command
    problem_grade = int(input())

    last_problem = problem_name
    total_score += problem_grade
    problems_count += 1

    if problem_grade <= 4:
        bad_grades_count += 1
        if bad_grades_count >= bad_grades_threshold:
            break

    command = input()

if command == "Enough":
    avg_score = total_score / problems_count
    print(f"Average score: {avg_score :.2f}")  # total_score / problems_count
    print(f"Number of problems: {problems_count}")
    print(f"Last problem: {last_problem}")
else:
    print(f"You need a break, {bad_grades_count} poor grades.")
